gtsfm/utils/images.py

```diff
@@ -118,6 +118,79 @@ def resize_image(image: Image, new_height: int, new_width: int) -> Image:
     return Image(resized_value_array)
 
 
+def get_rescaling_factor_per_axis(img_h: int, img_w: int, max_resolution: int) -> Tuple[float, float, int, int]:
+    """Resize an image such that the shorter image side is *exactly equal* to max_resolution.
+
+    Note: this may involve downsampling OR upsampling the image.
+
+    Resizing an image by a specific downsample ratio may not be possible due to lack of a clean
+    divisor. However, we can still determine the exact downsampling ratio.
+
+    Args:
+        img_h: height of image to be resized, in pixels
+        img_w: width of image to be resized, in pixels
+        max_resolution: integer representing length of image's short side
+            e.g. for 1080p (1920 x 1080), max_resolution would be 1080
+
+    Returns:
+        scale_u: rescaling factor for u coordinate
+        scale_v: rescaling factor for v coordinate. May not be equal to scale_u due to integer-rounding.
+        new_h: new height that will preserve aspect ratio as closely as possible, while
+            respecting max_resolution constraint
+        new_w: new width
+    """
+    h, w = img_h, img_w
+    shorter_size = min(h, w)
+    if shorter_size == h:
+        new_h = max_resolution
+        # compute scaling that will be applied to original image
+        scale = new_h / float(h)
+        new_w = np.round(w * scale)
+    else:
+        new_w = max_resolution
+        scale = new_w / float(w)
+        new_h = np.round(h * scale)
+
+    scale_u = new_w / w
+    scale_v = new_h / h
+
+    return scale_u, scale_v, new_h, new_w
+
+
+def get_downsampling_factor_per_axis(img_h: int, img_w: int, max_resolution: int) -> Tuple[float, float, int, int]:
+    """Resize an image such that the shorter image side is *less than or equal to* the max_resolution.
+
+    This will always downsample the image or leave it intact.
+
+    Note: Different from COLMAP's `Downsize()`, which instead checks if ANY side is
+        larger than the max resolution.
+        See: https://github.com/colmap/colmap/blob/dev/src/mvs/image.cc#L83
```

Comment:
do we need this note?

Reply:
i think it is a bit of a nice parallel with our approach vs. other lib, especially if we decide to change it later, so it might be useful to keep in mind/ keep as a note. But not required.